js/modules/k6/experimental/fs/registry.go

```diff
@@ -0,0 +1,60 @@
+package fs
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"sync"
+
+	"github.com/spf13/afero"
+)
+
+// registry is a registry of opened files.
+type registry struct {
+	// files holds a safe for concurrent use map of opened files.
+	//
+	// Keys are expected to be strings holding the files' path.
+	// Values are expected to be byte slices holding the files' data.
+	//
+	// That way, we can cache the file's content and avoid opening too many
+	// file descriptor, and re-reading its content every time the file is opened.
+	//
+	// Importantly, this also means that if the
+	// file is modified from outside of k6, the changes will not be reflected in the file's data.
+	// files map[string][]byte
+	files sync.Map
+}
+
+// open opens the named file for reading.
```

Comment:
Here I have a small confusion :thinking: 

Looking at the code I'd say that the current implementation of the `open`, does not exactly open the file, but reads and returns the file data with in-memory caching

Reply:
You're right, I'll adjust the docs to make the behavior more explicit indeed 👍🏻

Reply:
I've updated the docs, let me know if you spot further improvements, and feel free to unresolved accordingly 🙇🏻